File: data-jpa/src/main/java/top/bettercode/summer/data/jpa/config/JpaExtProperties.java
package top.bettercode.summer.data.jpa.config;

/**
 * @author Peter Wu
 */
public class JpaExtProperties {

  private SoftDelete softDelete = new SoftDelete();

  public SoftDelete getSoftDelete() {
    return softDelete;
  }

  public void setSoftDelete(SoftDelete softDelete) {
    this.softDelete = softDelete;
  }

  public static class SoftDelete {

    /**
     * 默认逻辑删除值.
     */
    private Object trueValue = 1;
    /**
     * 默认逻辑未删除值.
     */
    private Object falseValue = 0;

    public Object getTrueValue() {
      return trueValue;
    }

    public void setTrueValue(Object trueValue) {
      this.trueValue = trueValue;
    }

    public Object getFalseValue() {
      return falseValue;
    }

    public void setFalseValue(Object falseValue) {
      this.falseValue = falseValue;
    }
  }
}

File: data-jpa/src/main/kotlin/top/bettercode/summer/data/jpa/config/JpaExtProperties.kt
package top.bettercode.summer.data.jpa.config

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * @author Peter Wu
 */
@ConfigurationProperties("summer.data.jpa")
open class JpaExtProperties {

    /**
     * 逻辑删除
     */
    var logicalDelete = LogicalDelete()

    class LogicalDelete {
        /**
         * 默认逻辑删除值.
         */
        var trueValue: Any = 1

        /**
         * 默认逻辑未删除值.
         */
        var falseValue: Any = 0
    }
}
